platforms/windows/SQLitePlugin/SQLitePlugin.cpp

```diff
@@ -0,0 +1,511 @@
+                    onFailure(errorMessage);
+                    return;
+                }
+
+                onSuccess("Database Deleted");
+            }
+        });
+
+
+    }
+
+    void SQLitePlugin::EchoStringValue(
+        EchoStringValueOptions options,
+        std::function<void(std::string)> onSuccess,
+        std::function<void(std::string)> onFailure) noexcept
+    {
+        SerialReactDispatcher.Post(
+            [
+                options{ std::move(options) },
+                onSuccess{ std::move(onSuccess) }
+            ]()
+        {
+            onSuccess(options.Value);
+        });
+    }
+
+    void SQLitePlugin::ExecuteSqlBatch(
+        ExecuteSqlBatchOptions options,
+        std::function<void(std::vector<JSValueObject>)> onSuccess,
+        std::function<void(std::string)> onFailure) noexcept
+    {
+        SerialReactDispatcher.Post(
+            [
+                options{ std::move(options) },
+                onSuccess{ std::move(onSuccess) },
+                onFailure{ std::move(onFailure) },
+                weak_this = std::weak_ptr(shared_from_this())
+            ]()
+        {
+            if (auto strongThis = weak_this.lock()) {
+                if (options.DBArgs.DBName == nullptr)
+                {
+                    onFailure("You must specify database path");
+                    return;
+                }
+
+                hstring absoluteDbPath;
+                absoluteDbPath = ResolveDbFilePath(to_hstring(options.DBArgs.DBName));
+
+                if (strongThis->OpenDBs.find(absoluteDbPath) == strongThis->OpenDBs.end())
+                {
+                    onFailure("No such database, you must open it first");
+                    return;
+                }
+
+                sqlite3* dbHandle = strongThis->OpenDBs[absoluteDbPath];
+                std::vector<JSValueObject> results;
+
+                for (auto& query : options.Executes)
+                {
+                    JSValue result;
+                    if (ExecuteQuery(dbHandle, query, result))
+                    {
+                        //success
+                        results.push_back(
+                            {
+                                { "qid", query.QID },
+                                { "type", "success" },
+                                { "result", std::move(result)}
+                            }
+                        );
+                    }
+                    else
+                    {
+                        //query failed
+                        results.push_back(
+                            {
+                                { "qid", query.QID },
+                                { "type", "error" },
+                                { "error", result.AsString()},
+                                { "result", result.AsString()}
+                            }
+                        );
+                    }
+                }
+                onSuccess(std::move(results));
```

Comment:
move the success callback to the if above?

Reply:
We call onSuccess unless it is a critical failure like the DB is not open  or not specified.

We can batch multiple requests, and some of them might succeed and some might fail, and then we push results into a vector and return that vector that is parsed on JS side.

In this case, onSuccess is called properly, even if some request failed. The vector element is going to contain the error if it failed.